pyccl/halos/massdef.py

```diff
@@ -98,8 +98,11 @@ class MassDef(CCLAutoRepr, CCLNamedClass):
     @warn_api(pairs=[("c_m_relation", "concentration")])
     def __init__(self, Delta, rho_type=None, *, concentration=None):
         # Check it makes sense
-        if isinstance(Delta, str) and Delta not in ["fof", "vir"]:
-            raise ValueError(f"Unknown Delta type {Delta}.")
+        if isinstance(Delta, str):
+            if Delta.isdigit():
+                Delta = int(Delta)
```

Comment:
It's mainly for our convenience, so that a mass definition can be insantiated using `MassDef.from_name("300c")`.
It takes less lines of code to convert a string to digit than splitting up the string, checking if it can be converted to digit, and then converting it to one.
Internally `"300c"` becomes 2 strings: `"300"` and `"c"`.

Reply:
In that case, don't force it to be an integer, I think.

Reply:
Unfortunately there is no string method that checks for floats, only for integers.
So `MassDef.from_name("350.4c")` won't work. I can make it work but it will be a couple more lines for very little benefit I reckon.